tests/Doctrine/Tests/ORM/Functional/Ticket/Ticket5848Test.php

```diff
@@ -0,0 +1,167 @@
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\Ticket5848Client'),
+            ));
+        } catch (\Exception $e) {
+            // Swallow all exceptions. We do not test the schema tool here.
+        }
+
+
+        $this->_em->getEventManager()->addEventListener(Events::preFlush, $this->listener);
+    }
+
+    public function testExplicitFlushWithOne()
+    {
+        $e = new Ticket5848Client();
+
+        $this->_em->persist($e);
+        $this->_em->flush($e);
+
+        self::assertSame($e, $this->collector->entities[0]);
+    }
+
+    public function testExplicitFlushWithMany()
+    {
+        $e1 = new Ticket5848Client();
+        $e2 = new Ticket5848Authentication();
+        $e3 = new Ticket5848Client();
+
+        $this->_em->persist($e1);
+        $this->_em->persist($e2);
+        $this->_em->persist($e3);
+        $this->_em->flush([$e1, $e2]);
+
+        self::assertSame($e1, $this->collector->entities[0]);
+        self::assertSame($e2, $this->collector->entities[1]);
+    }
+
+    public function testExplicitFlushWithConsecutiveFlushes()
+    {
+        $e1 = new Ticket5848Client();
+        $e2 = new Ticket5848Authentication();
+        $e3 = new Ticket5848Client();
+
+        $this->_em->persist($e1);
+        $this->_em->persist($e2);
+        $this->_em->persist($e3);
+        $this->_em->flush([$e1, $e2]);
+        $this->_em->flush($e3);
+
+        self::assertSame($e3, $this->collector->entities[0]);
+    }
+
+    public function testImplicitFlush()
+    {
+        $e1 = new Ticket5848Client();
+        $e2 = new Ticket5848Authentication();
+        $e3 = new Ticket5848Client();
+
+        $this->_em->persist($e1);
+        $this->_em->persist($e2);
+        $this->_em->persist($e3);
+        $this->_em->flush();
+
+        self::assertSame($e1, $this->collector->entities[0]);
+        self::assertSame($e2, $this->collector->entities[1]);
+        self::assertSame($e3, $this->collector->entities[2]);
+    }
+
+    public function testExplicitFlushAfterImplicitFlush()
+    {
+        $e1 = new Ticket5848Client();
+        $e2 = new Ticket5848Authentication();
+        $e3 = new Ticket5848Client();
+
+        $this->_em->persist($e1);
+        $this->_em->persist($e2);
+        $this->_em->persist($e3);
+        $this->_em->flush();
+
+        self::assertSame($e1, $this->collector->entities[0]);
+        self::assertSame($e2, $this->collector->entities[1]);
+        self::assertSame($e3, $this->collector->entities[2]);
+
+        $e2->username = 'foo';
+
+        $this->_em->flush([$e2, $e3]);
+
+        self::assertSame($e2, $this->collector->entities[0]);
+        self::assertSame($e3, $this->collector->entities[1]);
+    }
+}
+
+class Ticket5848PreFlushListener
+{
+    public $collector;
+
+    public function preFlush(PreFlushEventArgs $event)
+    {
+        if (!$event->hasEntities()) {
+            $uow = $event->getEntityManager()->getUnitOfWork();
+            $uow->computeChangeSets();
```

Comment:
It does but that is just this particular use case. Having the entity list (or empty list) here will open up a whole new range of possibilities that could be done before doctrine actually processes the change set.